pyuvdata/uvdata/uvdata.py

```diff
@@ -12219,6 +12290,7 @@ def read(
         antenna_names=None,
         ant_str=None,
         bls=None,
+        catalog_names=None,
```

Comment:
Yeah, I think that would be easy enough to tackle -- certainly for `uvh5` at a minimum, and I'll take a quick look at `uvfits`.

Reply:
Okay, added to both `uvh5` and `uvfits`. Adding it to `ms` is a bit trickier since there's actually no _current_ infrastructure for doing any sort of "select-on-read" (due in part to MS/casacore having it's own SQL-like language for doing such things). That's going to be a bigger job, so I'd favor making that into an issue separate from this PR.

Reply:
Issue added -- #1283